astropy/units/core.py

```diff
@@ -521,6 +522,8 @@ def add_enabled_equivalencies(equivalencies):
         list of equivalent pairs, e.g., as returned by
         `~astropy.units.dimensionless_angles`.
     """
+    if not len(equivalencies):
```

Comment:
I think it is risky to make the change here, since people may specifically want to use this as a safe way to have a new context. But I'm totally in favour of doing the check in the decorator (around line 311, where `add_enabled_equivalencies` is used; i.e., omit it when `self.equivalencies` is empty)